tools/preflight/preflight.go

```diff
@@ -0,0 +1,620 @@
+package preflight
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	goexec "os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/trilioData/tvk-plugins/internal"
+	"github.com/trilioData/tvk-plugins/internal/utils/shell"
+	"github.com/trilioData/tvk-plugins/tools/preflight/exec"
+	"github.com/trilioData/tvk-plugins/tools/preflight/wait"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+// Options input options required for running preflight.
+type Options struct {
+	Context              context.Context
+	StorageClass         string
+	SnapshotClass        string
+	Kubeconfig           string
+	Namespace            string
+	LocalRegistry        string
+	ImagePullSecret      string
+	ServiceAccountName   string
+	PerformCleanupOnFail bool
+	Logger               *logrus.Logger
+}
+
+func InitKubeEnv(kubeconfig string) error {
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	kubeEnv, err := internal.NewEnv(kubeconfig, scheme)
+	clientSet = kubeEnv.GetClientset()
+	runtimeClient = kubeEnv.GetRuntimeClient()
+	discClient = kubeEnv.GetDiscoveryClient()
+	restConfig = kubeEnv.GetRestConfig()
+	return err
+}
+
+// createResourceNameSuffix creates a unique 6-length hash for preflight check.
+// All resources name created during preflight will have hash as suffix
+func createResourceNameSuffix() string {
+	suffix := make([]byte, 6)
+	randRange := big.NewInt(int64(len(letterBytes)))
+	for i := range suffix {
+		randNum, _ := rand.Int(rand.Reader, randRange)
+		idx := randNum.Int64()
+		suffix[i] = letterBytes[idx]
+	}
+
+	return string(suffix)
+}
+
+// PerformPreflightChecks performs all preflight checks.
+func (o *Options) PerformPreflightChecks() {
+	preflightStatus := true
+	resNameSuffix = createResourceNameSuffix()
+	storageSnapshotSuccess := true
+	var err error
+
+	o.Logger.Infoln(fmt.Sprintf("Generated UID for preflight check - %s", resNameSuffix))
+
+	//  check kubectl
+	o.Logger.Infoln("Checking for kubectl")
+	err = o.checkKubectl()
+	if err != nil {
+		o.Logger.Errorln(fmt.Sprintf("%s Preflight check for kubectl utility failed :: %s", cross, err.Error()))
+		preflightStatus = false
+	} else {
+		o.Logger.Infoln(fmt.Sprintf("%s Preflight check for kubectl utility is successful", check))
```

Comment:
use color to distinguish such messages.. check https://github.com/Sirupsen/logrus#formatters

also, ANSI char used for `check` and `cross` may not work on windows.. check once